src/components/hooks.js

```diff
@@ -33,13 +39,19 @@ export function useEditable(
   const [state, dispatch] = useReducer(reducer, initialState)
 
   const startEditing = () => dispatch({ type: types.START_EDITING })
+  const startUploading = () => dispatch({ type: types.START_UPLOADING })
```

Comment:
I think you can create a custom hooks that called `useEditable` and merges it into another reducer with the added special IPFS actions and states.